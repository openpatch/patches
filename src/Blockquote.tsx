/** @jsx jsx */
import { ReactNode, forwardRef, Ref } from "react";
import { jsx, css } from "@emotion/react";
import { TextProps } from "./Text";
import { Box } from "./Box";

export type BlockquoteProps = {
  children?: ReactNode;
} & Omit<TextProps, "as">;

const Icon = () => {
  return (
    <Box
      position="absolute"
      css={css`
        margin-left: -0.5rem;
        margin-top: -0.5rem;
      `}
    >
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 30 30"
        css={(theme) => css`
          fill: ${theme.colors.neutral[100]};
          width: 2rem;
          height: 2rem;
        `}
      >
        <path d="M15.264 19.552c0 3.648 2.4 5.568 5.088 5.568 2.976 0 5.76-2.496 5.76-5.664 0-2.592-1.824-4.32-4.032-4.32h-.768c1.056-3.072 4.512-6.336 8.064-8.064L26.4 4c-5.952 2.976-11.136 9.504-11.136 15.552zM0 19.552C0 23.2 2.304 25.12 5.088 25.12c2.976 0 5.76-2.496 5.76-5.664 0-2.592-1.92-4.32-4.128-4.32h-.768c1.056-3.072 4.608-6.336 8.064-8.064L11.136 4C5.184 6.976 0 13.504 0 19.552z"></path>
      </svg>
    </Box>
  );
};

export default Icon;

export const Blockquote = ({ children, ...props }: BlockquoteProps) => {
  return (
    <Box as="blockquote" fontStyle="italic">
      <Icon />
      <Box position="relative">
        <p>{children}</p>
      </Box>
    </Box>
  );
};
