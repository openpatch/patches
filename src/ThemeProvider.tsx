import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
