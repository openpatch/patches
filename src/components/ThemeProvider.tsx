import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
