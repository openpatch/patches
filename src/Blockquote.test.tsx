import { render } from "@testing-library/react";
import { Blockquote, BlockquoteProps } from "./Blockquote";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: BlockquoteProps) =>
  render(
    <ThemeProvider>
      <Blockquote {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
