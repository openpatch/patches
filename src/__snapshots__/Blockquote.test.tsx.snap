// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  font-style: italic;
  position: relative;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  position: absolute;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.emotion-4 {
  fill: #E4E7EB;
  width: 2rem;
  height: 2rem;
}

.emotion-5 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  position: relative;
}

<div>
  <blockquote
    class="emotion-0 emotion-1"
    font-style="italic"
  >
    <div
      class="emotion-2 emotion-1"
    >
      <svg
        class="emotion-4"
        fill="none"
        viewBox="0 0 30 30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.264 19.552c0 3.648 2.4 5.568 5.088 5.568 2.976 0 5.76-2.496 5.76-5.664 0-2.592-1.824-4.32-4.032-4.32h-.768c1.056-3.072 4.512-6.336 8.064-8.064L26.4 4c-5.952 2.976-11.136 9.504-11.136 15.552zM0 19.552C0 23.2 2.304 25.12 5.088 25.12c2.976 0 5.76-2.496 5.76-5.664 0-2.592-1.92-4.32-4.128-4.32h-.768c1.056-3.072 4.608-6.336 8.064-8.064L11.136 4C5.184 6.976 0 13.504 0 19.552z"
        />
      </svg>
    </div>
    <div
      class="emotion-5 emotion-1"
    />
  </blockquote>
</div>
`;
